cpp/core/shuffle/PartitionWriter.h

```diff
@@ -21,17 +21,36 @@
 
 namespace gluten {
 
+class SpillHandle {
+ public:
+  virtual ~SpillHandle() = default;
+
+  virtual arrow::Status consume(uint32_t partitionId, std::unique_ptr<arrow::ipc::IpcPayload> payload) = 0;
+
+  virtual arrow::Status close() = 0;
+
+  virtual bool closed() = 0;
+};
+
 class ShuffleWriter::PartitionWriter {
  public:
   PartitionWriter(ShuffleWriter* shuffleWriter) : shuffleWriter_(shuffleWriter) {}
   virtual ~PartitionWriter() = default;
 
   virtual arrow::Status init() = 0;
 
-  virtual arrow::Status evictPartition(int32_t partitionId) = 0;
-
   virtual arrow::Status stop() = 0;
 
+  /// Request next spill.
+  /// \param flushOnSpill Whether to flush the spilled data immediately. If it's false,
+  /// the spilled data can be cached first.
+  virtual arrow::Status requestNextSpill(bool flushOnSpill) = 0;
+
+  /// Get the current managed SpillHandle. Returns nullptr if the current SpillHandle was closed,
+  /// or requestNextSpill has not been called.
+  /// \return
+  virtual std::shared_ptr<SpillHandle> getSpillHandle() = 0;
+
```

Comment:
Was there any reason preventing using a single member function `virtual Result<std::shared_ptr<SpillHandle>> requestNextSpill`?

Reply:
The caller can use `requestNextSpill` to start a spill, and choose to call `getSpillHandle()->consume()` immediately, or to call it latter somewhere else. Only `requestNextSpill` cannot make the caller to start a new spill multiple times.